Unable to install pnpm from corepack in the current repo

### Verify latest release

- [x] I verified that the issue exists in the latest pnpm release

### pnpm version

11.0.0-alpha.1

### Which area(s) of pnpm are affected? (leave empty if unsure)

_No response_

### Link to the code that reproduces this issue or a replay of the bug

_No response_

### Reproduction steps

1. Clone the pnpm repo
2. Run any `pnpm` command in the repo

### Describe the Bug

Corepack tries to run `pnpm.cjs`, but the compiled file is now `pnpm.mjs`:

```sh
pnpm compile       
! Corepack is about to download https://registry.npmjs.org/pnpm/-/pnpm-11.0.0-alpha.1.tgz
? Do you want to continue? [Y/n] Y

node:internal/modules/cjs/loader:1404
  throw err;
  ^

Error: Cannot find module '/Users/trevorbb/.cache/node/corepack/v1/pnpm/11.0.0-alpha.1/bin/pnpm.cjs'
    at Function._resolveFilename (node:internal/modules/cjs/loader:1401:15)
    at defaultResolveImpl (node:internal/modules/cjs/loader:1057:19)
    at resolveForCJSWithHooks (node:internal/modules/cjs/loader:1062:22)
    at Function._load (node:internal/modules/cjs/loader:1211:37)
    at TracingChannel.traceSync (node:diagnostics_channel:322:14)
    at wrapModuleLoad (node:internal/modules/cjs/loader:235:24)
    at executeUserEntryPoint (node:internal/modules/run_main:171:5)
    at process.processTicksAndRejections (node:internal/process/task_queues:89:21) {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}
```

### Expected Behavior

The version of pnpm installed by Corepack should work.

I thought maybe there was a discrepancy in the Corepack cache, but I get the same behavior after running `corepack cache clean`.

Doing something like copying `pnpm.mjs` to `pnpm.cjs` doesn't work, since Node can't parse it correctly with the mismatched extension.

### Which Node.js version are you using?

24.6.0

### Which operating systems have you used?

- [x] macOS
- [ ] Windows
- [ ] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

_No response_

--- a/pnpm/bin/pnpm.cjs
+++ b/pnpm/bin/pnpm.cjs
@@ -1,3 +1,3 @@
 #!/usr/bin/env node
 // CommonJS shim for compatibility with older Corepack versions
-import("./pnpm.mjs");
+import('./pnpm.mjs')

--- a/pnpm/bin/pnpx.cjs
+++ b/pnpm/bin/pnpx.cjs
@@ -1,3 +1,3 @@
 #!/usr/bin/env node
 // CommonJS shim for compatibility with older Corepack versions
-import("./pnpx.mjs");
+import('./pnpx.mjs')

